bin/BuildPackages.sh

```diff
@@ -79,115 +135,197 @@ build_carat() {
 # It is not possible to move around compiled binaries because these have the
 # path to some data files burned in.
 zcat carat-2.1b1.tgz | tar pxf -
-ln -s carat-2.1b1/bin bin
+# If the symbolic link does not exist then create it.
+# Does not check if bin exists already with some content or as a file, etc.
+# Thus this code is unstable, but should be corrected by package author
+# and not by this script.
+if [[ ! -L ./bin ]]
+then
+  ln -s ./carat-2.1b1/bin ./bin
+fi
 cd carat-2.1b1
-make TOPDIR=`pwd`
+make TOPDIR="$(pwd)"
 chmod -R a+rX .
 cd bin
-aa=`./config.guess`
-for x in "`ls -d1 $GAPDIR/bin/${aa}*`"; do
- ln -s "$aa" "`basename $x`"
-done
+# This sets GAParch as it should be.
+# Alternatively, one could check $GAPDIR/bin for all directories instead.
+source "$GAPDIR/sysinfo.gap"
+# If the symbolic link does not exist then create it.
+if [[ ! -L ./"$GAParch" ]]
+then
+  shopt -s nullglob
+  # We assume that there is only one appropriate directory....
+  for archdir in *
+  do
+    if [[ -d "$archdir" ]] && [[ ! -L "$archdir" ]]
+    then
+      ln -s ./"$archdir" ./"$GAParch"
+    fi
+  done
+  shopt -u nullglob
+fi
```

Comment:
Hrm, but the old script also had a flaw: it was not idempotent.

And reading it, and looking at what carat does, I doubt the old script worked correctly: It does `ln -s "$aa" "`basename $x`"` where `aa` is the `config.guess` output -- so it does *not* contain the name of the C compiler. But the carat Makefile does insert that...

I'll try to write a proper replacement. Just got carat to compile, but it required quite some effort...